databuilder/query_engines/base_sql.py

```diff
@@ -428,7 +498,10 @@ def get_element_from_value_from_function(
         #
         # but the simple thing will do for now. Note we can't use `singledispatchmethod`
         # for this because it doesn't play nicely with subclassing.
-        class_method_map = {
+        class_method_map: dict[
+            type[ValueFromFunction],
+            Union[Callable[[Any], ClauseElement], Callable[[Any, Any], ClauseElement]],
+        ] = {
```

Comment:
We're promising the type checker that the dict really does provide mappings for all subclasses of `ValueFromFunction`. The type checker's uncertainty here reflects the reality that we have to remember to add a new entry to the dict when we add a new function.